angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
What about simply this?
```ts
<T extends null | undefined>(array: T, expression?: string | ((value: any) => any) | (((value: any) => any) | string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T;
<T>(array: T[], expression?: string | ((value: T) => any) | (((value: T) => any) | string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Or should we use something else in place of `((value: any) => any)`? Probably `never` somehow? So that the compiler could tell that it doesn't make sense to pass a function that will never be called?